### ISSUE TITLE
Windows XP is unable to execute 32-bit exe program

### ISSUE THREAD

**Issue by bingbian001** [fvarrui/JavaPackager#350]
Windows XP is unable to execute 32-bit exe program. Please ask how to resolve this issue.
 Thank you.

**Comment by fvarrui** [comment]
Did you bundle a 32 bit JRE with your app?

**Comment by bingbian001** [comment]
Yes, I have bundled 32-bit jre into exe, but the xp system still prompts that it is not a valid 32-bit program, but I solved this problem with exe4j.

**Comment by bingbian001** [comment]
> Did you bundle a 32 bit JRE with your app?

Yes, I have bundled 32-bit jre into exe, but the xp system still prompts that it is not a valid 32-bit program, but I solved this problem with exe4j.

**Comment by bingbian001** [comment]
> Did you bundle a 32 bit JRE with your app?

Yes, I have bundled 32-bit jre into app, but the xp system still prompts that it is not a valid 32-bit program.
My JDK version is jdk-8u1710-windows-i586, and my XP version is service pack 3.
However, I was able to successfully install and use the same JDK on the XP system by running exe on exe4j.

**Comment by bingbian001** [comment]
> Did you bundle a 32 bit JRE with your app?

I seem to have found the reason and used Inno Setup 6.2.2 during the packaging process. It seems that versions of Inno Setup 6 and above do not support installation on the xp system and require Inno Setup 5 and below.

**Comment by fvarrui** [comment]
So, was the problem with Setup installer instead of your app executable?

**Comment by fvarrui** [comment]
> I seem to have found the reason and used Inno Setup 6.2.2 during the packaging process. It seems that versions of Inno Setup 6 and above do not support installation on the xp system and require Inno Setup 5 and below.

Ok, maybe updating docs would be enough?

**Comment by bingbian001** [comment]
> So, was the problem with Setup installer instead of your app executable?

Yes, the application is executable, it's just a matter of installing the program

**Comment by bingbian001** [comment]
> So, was the problem with Setup installer instead of your app executable?

Yes, the application is executable, it's just a matter of installing the program

**Comment by bingbian001** [comment]
> > I seem to have found the reason and used Inno Setup 6.2.2 during the packaging process. It seems that versions of Inno Setup 6 and above do not support installation on the xp system and require Inno Setup 5 and below.
> 
> Ok, maybe updating docs would be enough?

I uninstalled Inno Setup 6. Inno Setup 5 is installed. Debugging the exe package There is currently a new error 
'Setup installer generation failed due to: command execution failed iscc'
I am trying to find the reason. However, after installing Inno Setup 6, I was able to successfully execute it again.

**Comment by fvarrui** [comment]
> Debugging the exe package There is currently a new error
> 'Setup installer generation failed due to: command execution failed iscc'

Inno Setup directory must be on PATH environment variable, so that `iscc` command can be run

**Comment by bingbian001** [comment]
> > Debugging the exe package There is currently a new error
> > 'Setup installer generation failed due to: command execution failed iscc'
> 
> Inno Setup directory must be on PATH environment variable, so that `iscc` command can be run

Thank you. I'll just reinstall and uninstall Inno Setup 5. Thank you for your help

**Comment by fvarrui** [comment]
Maybe this issue could be related to https://github.com/fvarrui/JavaPackager/issues/338#issuecomment-1647990618 and the new `arch` option could fix the problem

**Comment by fvarrui** [comment]
Hi @bingbian001!
Can we close this issue?
